### PR TITLE
Release v18 boojum

### PR DESCRIPTION
**Pull request by StanislavBreadless**
# What ❔

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

### CODE REVIEW

File: contracts/Compressor.sol [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    /// @dev The compression algorithm works as follows:
+    ///     1. The original bytecode is split into 8-byte chunks.
+    ///     Since the bytecode size is always a multiple of 32, this is always possible.
+    ///     2. For each 8-byte chunk in the original bytecode:
+    ///         * If the chunk is not already in the dictionary, it is added to the dictionary array.
+    ///         * If the dictionary becomes overcrowded (2^16 + 1 elements), the compression process will fail.
+    ///         * The 2-byte index of the chunk in the dictionary is added to the encoded data.
+    /// @dev Currently, the method may be called only from the bootloader because the server is not ready to publish bytecodes
+    /// in internal transactions. However, in the future, we will allow everyone to publish compressed bytecodes.
+    function publishCompressedBytecode(
```
Comment (vladbochok) [inline contracts/Compressor.sol:42]: I don't think we added `require(_bytecode.length >= _rawCompressedData.length);`

File: bootloader/tests/utils/test_utils.yul [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+function testing_totalTests(tests) {
+    storeTestHookParam(0, unoptimized(tests))
+    setTestHook(nonOptimized(103))
+}
+
+// Returns txDataOffset for the index transaction.
+function testing_txDataOffset(index) -> txDataOffset {
+    let txPtr := add(TX_DESCRIPTION_BEGIN_BYTE(), mul(index, TX_DESCRIPTION_SIZE()))
+    txDataOffset := mload(add(txPtr, 0x20))
+}
```
Comment (vladbochok) [inline bootloader/tests/utils/test_utils.yul:54]: Newline

File: bootloader/tests/bootloader/bootloader_test.yul [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    testing_assertEq(getStrLen("12345678901234567890123456789012"), 32, "max length")
+    testing_assertEq(getStrLen("1234567890123456789012345678901234"), 0, "over max length")
+}
+
+function TEST_simple_transaction() {
+    // We'll test the transaction from 0.json
+    let txDataOffset := testing_txDataOffset(0)
+    let innerTxDataOffset := add(txDataOffset, 0x20)
+    testing_assertEq(getGasPerPubdataByteLimit(innerTxDataOffset), 0xc350, "Invalid pubdata limit")
+}
```
Comment (vladbochok) [inline bootloader/tests/bootloader/bootloader_test.yul:52]: newline

File: contracts/Compressor.sol (hunk 1 above)
Comment (StanislavBreadless) [reply contracts/Compressor.sol:42]: We'll add it in the next release

**Review by vladbochok — APPROVED** [review]

**Review by benceharomi — APPROVED** [review]

**Pull request merged**